### PR TITLE
Extend C Isogram Test Suite

### PR DESCRIPTION
**Pull request by FroshVII**
- Added test for NULL input
- Added test for duplicated letters on the ends of character arrays
- ~~Added test for terminating as soon as a duplicate is found~~

### CODE REVIEW

**Comment by wolf99** [comment]
Thanks for the work @FroshVII, bu are these tests part of the problems specifications? (https://github.com/exercism/problem-specifications/blob/master/exercises/isogram/canonical-data.json)

Also, IIRC testing for NULL cases was actually deemed to be unrequired for exercism problem sets (can't think where I read this here so this may indeed be recalled incorrectly, I'll link it if I think of it).

**Comment by FroshVII** [comment]
@wolf99 Well the test for duplicated characters was inspired by @wiredancer's comment at the top of those specs, but the other two are admittedly just additions I personally want confirmed in my code.

**Comment by wolf99** [comment]
Seems the discussion on invalid input does not have a conclusion  (though this is not the discussion I am trying to recall) https://github.com/exercism/problem-specifications/blob/master/exercises/isogram/canonical-data.json

**Comment by wolf99** [comment]
FYI, clock time from the C library on any given system can have many influencing factors in comparing between any two runs, such that for single occurrences of very short times (such as this implementation) can often give incorrect results. When comparing measurements of execution time programmers usually record the time it takes to run several tens, if not hundreds, of executions.

File: exercises/isogram/test/test_isogram.c [hunk 1]
```diff
@@ -15,6 +15,12 @@ void test_empty_string(void)
...
 
+void test_null(void)
+{
+   TEST_IGNORE();
+   TEST_ASSERT_FALSE(is_isogram(NULL));
+}
+
 void test_lower_case_only(void)
 {
    TEST_IGNORE();               // delete this line to run test
```
Comment (ryanplusplus) [inline exercises/isogram/test/test_isogram.c]: Can you move `// delete this line to run test` to the new test? We put this on the second test of all exercises.

Comment (ryanplusplus) [reply exercises/isogram/test/test_isogram.c]: Since you have your settings configured to allow me to make changes on your pull requests for this repository (thanks!) I went ahead and made this small change.

**Comment by ryanplusplus** [comment]
@wolf99 Discussion regarding invalid inputs is ongoing, but for now it looks like there's precedent on this track for checking for a `NULL` string input (https://github.com/exercism/c/blob/master/exercises/pangram/test/test_pangram.c#L12) so let's go ahead with it for now and come back if the discussion resolves.

**Comment by ryanplusplus** [comment]
Thanks @FroshVII for your contribution!
